tests/encoding_circuit/test_pruned_encoding_circuit.py

```diff
@@ -32,5 +34,12 @@ def test_get_circuit(self):
 
         assert isinstance(qc, QuantumCircuit)
         assert qc.num_qubits == 4
-        # encdoding slots got pruned
+        # encoding slots got pruned
         assert pruned.num_encoding_slots == circuit.num_encoding_slots - 4
+
+    def test_automated_pruning(self):
+        circuit = LayeredEncodingCircuit.from_string("Rz(x)-crz(p)-Ry(x)-crx(p)", num_qubits=4)
+        circuit._build_layered_pqc(4)
+        pruned = automated_pruning(circuit, Executor())
+        assert isinstance(pruned, PrunedEncodingCircuit)
+        assert pruned.num_parameters == circuit.num_parameters - len(pruned._pruned_parameters)
```

Comment:
This is a bit better but still I would prefer a test that compares the output circuit to one you would expect